app/services/validator_check_active_service.rb

```diff
@@ -27,6 +29,16 @@ def update_validator_activity
 
   private
 
+  def should_be_destroyed?(validator)
+    if !validator.validator_histories.where("created_at > ?", @delinquent_time.ago).exists?
+      if validator.validator_histories.where('created_at < ?', @delinquent_time.ago).exists?
+        return true
+      end
+    end
+
+    false
+  end
+
   # number of previous by network
   def previous_epoch(network)
     current_epoch = EpochWallClock.where(network: network).order(created_at: :desc).last
```

Comment:
If it's not intentionally this also could be cached to not generate db queries every time.